deployment/ccip/changeset/cs_deploy_token_pools_test.go

```diff
@@ -0,0 +1,689 @@
+						},
+						Type:               BurnMintTokenPool,
+						LocalTokenDecimals: localTokenDecimals,
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	state, err := LoadOnchainState(e)
+	require.NoError(t, err)
+
+	for _, selector := range []uint64{selectorA, selectorB} {
+		validateMemberOfBurnMintPair(t, state, tokens, testTokenSymbol, selector, 100, 1000, e.Chains[selector].DeployerKey)
+	}
+}
+
+func TestDeployTokenPoolContracts_KeepExistingWithTransferToExternalAdmin(t *testing.T) {
+	t.Parallel()
+	e, selectorA, selectorB, tokens, timelockContracts := setup2ChainEnvironment(t)
+
+	// Initial deployment
+	e, err := commonchangeset.ApplyChangesets(t, e, timelockContracts, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(DeployTokenPoolContracts),
+			Config: DeployTokenPoolContractsConfig{
+				Symbol:        testTokenSymbol,
+				TimelockDelay: 0 * time.Second,
+				NewPools: map[uint64]NewTokenPoolInput{
+					selectorA: {
+						BaseTokenPoolInput: BaseTokenPoolInput{
+							TokenAddress: tokens[selectorA].Address,
+							RemoteChainsToAdd: RemoteChains{
+								selectorB: createSymmetricRateLimits(100, 1000),
+							},
+						},
+						Type:               BurnMintTokenPool,
+						LocalTokenDecimals: localTokenDecimals,
+					},
+					selectorB: {
+						BaseTokenPoolInput: BaseTokenPoolInput{
+							TokenAddress: tokens[selectorB].Address,
+							RemoteChainsToAdd: RemoteChains{
+								selectorA: createSymmetricRateLimits(100, 1000),
+							},
+						},
+						Type:               BurnMintTokenPool,
+						LocalTokenDecimals: localTokenDecimals,
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	state, err := LoadOnchainState(e)
+	require.NoError(t, err)
+
+	for _, selector := range []uint64{selectorA, selectorB} {
+		validateMemberOfBurnMintPair(t, state, tokens, testTokenSymbol, selector, 100, 1000, nil)
+	}
+
+	// Transfer existing pools to an external administrator
+	e, err = commonchangeset.ApplyChangesets(t, e, timelockContracts, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(DeployTokenPoolContracts),
+			Config: DeployTokenPoolContractsConfig{
+				Symbol:        testTokenSymbol,
+				TimelockDelay: 0 * time.Second,
+				ExistingPoolUpdates: map[uint64]BaseTokenPoolInput{
+					selectorA: {
+						TokenAddress:  tokens[selectorA].Address,
+						ExternalAdmin: e.Chains[selectorA].DeployerKey.From,
+					},
+					selectorB: {
+						TokenAddress:  tokens[selectorB].Address,
+						ExternalAdmin: e.Chains[selectorB].DeployerKey.From,
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	state, err = LoadOnchainState(e)
+	require.NoError(t, err)
+
+	for _, selector := range []uint64{selectorA, selectorB} {
+		validateMemberOfBurnMintPair(t, state, tokens, testTokenSymbol, selector, 100, 1000, e.Chains[selector].DeployerKey)
+	}
+}
+
+func TestDeployTokenPoolContracts_RedeployNew(t *testing.T) {
+	/*
```

Comment:
I believe this should be broken off into a separate ticket, as it could change the on-chain state struct and therefore break other consumers.